app/src/main/java/org/liberty/android/fantastischmemo/ui/QuizHistoryActivity.java

```diff
@@ -0,0 +1,68 @@
+package org.liberty.android.fantastischmemo.ui;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+import org.liberty.android.fantastischmemo.R;
+import org.liberty.android.fantastischmemo.entity.History;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.TimeZone;
+
+public class QuizHistoryActivity extends AppCompatActivity {
+
+    public static final String EXTRA_DBPATH = "dbpath";
+
+    private TextView averageTextView;
+    private TextView attemptTextView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.quiz_history_activity);
+
+        ArrayList<History> histories = this.getIntent().getParcelableArrayListExtra("HISTORY");
+        averageTextView = (TextView) findViewById(R.id.quiz_history_average);
+        attemptTextView = (TextView) findViewById(R.id.quiz_history_attempt);
+
+        if (histories.size()>0) {
+            Double average = computeAverage(histories);
+
+            averageTextView.setText("Average: " + average + "\n\n");
+
+            int attempt = 1;
+            for(History history : histories){
+                String estDate = estDate(history.getTimeStamp());
+                attemptTextView.append("Attempt " + attempt + ": " + history.getMark() + "% (" + estDate + ")\n\n");
+                attempt++;
+            }
+
+        }
+        else {
+            attemptTextView.setText("You have not attempted this quiz!");
+        }
+
+    }
+
+    private Double computeAverage(ArrayList<History> histories) {
+        double total = 0;
+        for(History history : histories){
+            total += history.getMark();
+        }
+        return Math.floor((total/histories.size())*100)/100;
+    }
+
+    private String estDate(Long timeStamp){
+        //Daylight savings offset
+        long offset = ((1*1000)*60)*60;
```

Comment:
do you mind explaining to me how the offset is calculated?

Reply:
Pretty much I am just converting an hour into milliseconds.
1*1000 = 1000 <-- milliseconds in a second
60*1000 = 60 000 <-- milliseconds in a minute
60 * 60 000 = 3 600 000 <-- milliseconds in an hour

I can just put the final number if you want, I just thought it would be clearer to show the calculations.

Reply:
okie cool! I think it's fine